packages/is-shallow-equal/objects.js

```diff
@@ -28,7 +28,12 @@ function isShallowEqualObjects( a, b ) {
 
 	while ( i < aKeys.length ) {
 		key = aKeys[ i ];
-		if ( a[ key ] !== b[ key ] ) {
+		aValue = a[ key ];
+
+		if (
+			( aValue === undefined && ! b.hasOwnProperty( key ) ) ||
```

Comment:
It'd be useful to add a comment here explaining why the comparison is being done like this.

Reply:
> It'd be useful to add a comment here explaining why the comparison is being done like this.

Agree 👍 Added in 741fc5c97.